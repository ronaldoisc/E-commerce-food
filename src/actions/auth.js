import { useFetch } from "../helpers/fetch"
import { types } from "../types/types";
import Swal from 'sweetalert2'

export const startLogin=(username,password)=>{
    
    return async(dispatch)=>{
        try {
            const resp=await useFetch('auth/login',{username,password},'POST');
            const body=await resp.json();
            if(!body.msg){
               localStorage.setItem('token',JSON.stringify({username,token:body}));
               dispatch(login({ username}));
            }else{
                Swal.fire('Error',body.msg,'error');
            }
        } catch (error) {
            Swal.fire('Error','Something was wrong!','error');
        }
    }
}

const login=(user)=>({
    type:types.authLogin,
    payload:user

})


export const startLogout=()=>{
    return (dispatch)=>{
        localStorage.clear();
        dispatch(logout());
        
    }
}

const logout=()=>({
    type:types.authLogout
});




export const startChecking=()=>{
    return (dispatch)=>{
        //check if exist a token
        const user=localStorage.getItem('token');
        if(user!=null){
           const {username}= JSON.parse(user);
            dispatch(login({username}))
        }else{
            dispatch(checkingFinish());
        }
    }
}


const checkingFinish=()=>({
    type:types.authCheckingFinish
})

