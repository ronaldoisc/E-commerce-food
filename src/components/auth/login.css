.container-login{
   background-color: rgb(231, 231, 235);
   height: 100vh;
   width: 100vw;
   display: flex;
   flex-direction: column;
   justify-content: center;
   align-items: center;
 

}
@media (min-width: 768px) {
    .container-login{
        height: 100vh;
        display: flex;
        flex-direction: row;
        justify-content: space-evenly;
        padding: 4rem;
        align-items: unset;
    }
    
}
.container--login__column__signUp{
    background-color: #ff7300;
    box-shadow: 10px 5px 5px rgb(216, 212, 212);
    width: 90%;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    padding: 0.5rem;
   

}
@media (min-width: 768px) {
    .container--login__column__signUp{
        width: 50%;
    }
}

.container--login__column__signUp p{
    color: white;
    font-size: 1rem;
}
.container-login__column__form{
  box-shadow: 10px 5px 5px rgb(216, 213, 213);
  align-items: center;
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 1rem;
  width: 90%;
}
@media (min-width: 768px) {
  .container-login__column__form{
      width: 50%;
  }  
    
}

.container-login__column__form input{
    margin-bottom: 1rem;
    padding: 0.5rem;
    width: 100%;
}
.container-login__column__form input:focus{
    outline: none;
}
.btn-login{
    border: none;
    outline: none;
    width: 100%;
}
.btn-login:hover{
    background-color: black !important;
}