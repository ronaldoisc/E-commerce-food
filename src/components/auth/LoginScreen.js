import React from 'react';
import { useDispatch } from 'react-redux';

import { startLogin } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';
import './login.css';

export const LoginScreen = () => {


  const dispatch = useDispatch();
  const [ formLoginValues, handleLoginInputChange] = useForm( {username:'', password:''
    });

const {username,password}=formLoginValues;



const handleLoginSubmit=(e)=>{
  e.preventDefault();
  dispatch(startLogin(username,password))

}

  return <div>
    <div className='container-login'>
      <div className='container--login__column__signUp'>
        <h1>Hi!,welcome</h1>
        <p>Go ahead get access to your account and buy something!</p>
      </div>
      <div className='container-login__column__form'>
        <h1>E-COMMERCE</h1>
        <form onSubmit={handleLoginSubmit}>

          <label htmlFor="username">Username</label>
          <input type="text" name='username' value={username} onChange={handleLoginInputChange} placeholder='write your email' required />
          <label htmlFor="password">Password</label>
          <input type="text" name='password' value={password}  onChange={handleLoginInputChange} placeholder='write your password' required />

          <button className='btn btn-login'>Login</button>
        </form>

      </div>

    </div>
  </div>;
};
