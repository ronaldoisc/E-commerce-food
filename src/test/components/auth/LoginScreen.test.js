import configureStore from 'redux-mock-store' //ES6 modules
import thunk from 'redux-thunk';

import {mount} from 'enzyme';
import { Provider } from 'react-redux';
import { LoginScreen } from '../../../components/auth/LoginScreen';



const middlewares = [thunk]
const mockStore = configureStore(middlewares);

const initialState = {}
const store = mockStore(initialState)

describe('test in the <LoginScreen/>', () => {
  
    test('should show correctly', async () => {
     
        const wrapper=mount(
            <Provider store={store}>
                <LoginScreen/>
            </Provider>
        )

        expect(wrapper).toMatchSnapshot();
      
    });
    
});
