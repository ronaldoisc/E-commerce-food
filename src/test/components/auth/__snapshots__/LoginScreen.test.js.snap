// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in the <LoginScreen/> should show correctly 1`] = `
<div>
  <div
    className="container-login"
  >
    <div
      className="container--login__column__signUp"
    >
      <h1>
        Hi!,welcome
      </h1>
      <p>
        Go ahead get access to your account and buy something!
      </p>
    </div>
    <div
      className="container-login__column__form"
    >
      <h1>
        E-COMMERCE
      </h1>
      <form
        onSubmit={[Function]}
      >
        <label
          htmlFor="username"
        >
          Email
        </label>
        <input
          name="username"
          onChange={[Function]}
          placeholder="write your email"
          required={true}
          type="text"
          value=""
        />
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          name="password"
          onChange={[Function]}
          placeholder="write your password"
          required={true}
          type="text"
          value=""
        />
        <button
          className="btn btn-login"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;
